src/components/item/index.tsx

```diff
@@ -25,18 +25,17 @@ export class SmoothlyItem implements Item, ComponentWillLoad, ComponentDidLoad {
 	@Prop({ reflect: true, mutable: true }) selected: boolean
 	@Prop({ reflect: true, mutable: true }) marked: boolean
 	@Prop() selectable = true
-	@Event() smoothlyItemSelect: EventEmitter<void>
+	@Event() smoothlyItemSelect: EventEmitter<HTMLSmoothlyItemElement>
 	@Event() smoothlyInputLoad: EventEmitter<(parent: HTMLElement) => void>
 	@Watch("selected")
-	onSelectedChanged(value: boolean, old: boolean) {
-		if (value && !old && this.selectable)
-			this.smoothlyItemSelect.emit()
+	onSelectedChanged() {
+		if (this.selectable)
+			this.smoothlyItemSelect.emit(this.element)
```

Comment:
This is fine but this is also available on the `event.target` you can use those interfaces we used last friday on friday fika to get the correct type for `event.target`